ethcore/src/state/mod.rs

```diff
@@ -186,8 +194,8 @@ impl AccountEntry {
 /// checkpoint can be discateded with `discard_checkpoint`. All of the orignal
 /// backed-up values are moved into a parent checkpoint (if any).
 ///
-pub struct State {
-	db: StateDB,
+pub struct State<B> {
```

Comment:
More general > less general. This is a common pattern among e.g. structures in `std`: constrain the generic parameters only when introducing functionality that depends on those constraints.